roomserver/internal/helpers/auth.go

```diff
@@ -131,6 +132,14 @@ type authEvents struct {
 	events         EventMap
 }
 
+func (ae *authEvents) Valid() bool {
+	roomIDs := make(map[string]struct{})
+	for _, ev := range ae.events {
+		roomIDs[ev.RoomID()] = struct{}{}
+	}
+	return len(roomIDs) <= 1
```

Comment:
This is inefficient as we'll loop through all events regardless. Just remember the first room id and return false if any event isn't that.

Reply:
Where is this function called from, and do we know the room ID at that point?